internal/exec/path_utils.go

```diff
@@ -0,0 +1,86 @@
+package exec
+
+import (
+	"fmt"
+	c "github.com/cloudposse/atmos/pkg/config"
+	"path"
+)
+
+// constructTerraformComponentWorkingDir constructs the working dir for a terraform component in a stack
+func constructTerraformComponentWorkingDir(info c.ConfigAndStacksInfo) string {
+	return path.Join(
+		c.Config.BasePath,
+		c.Config.Components.Terraform.BasePath,
+		info.ComponentFolderPrefix,
+		info.FinalComponent,
+	)
+}
+
+// constructTerraformComponentPlanfileName constructs the planfile name for a terraform component in a stack
+func constructTerraformComponentPlanfileName(info c.ConfigAndStacksInfo) string {
+	var planFile string
+	if len(info.ComponentFolderPrefix) == 0 {
+		planFile = fmt.Sprintf("%s-%s.planfile", info.ContextPrefix, info.Component)
+	} else {
+		planFile = fmt.Sprintf("%s-%s-%s.planfile", info.ContextPrefix, info.ComponentFolderPrefix, info.Component)
+	}
+
+	return planFile
+}
+
+// constructTerraformComponentVarfileName constructs the varfile name for a terraform component in a stack
+func constructTerraformComponentVarfileName(info c.ConfigAndStacksInfo) string {
+	var varFile string
+	if len(info.ComponentFolderPrefix) == 0 {
+		varFile = fmt.Sprintf("%s-%s.terraform.tfvars.json", info.ContextPrefix, info.Component)
+	} else {
+		varFile = fmt.Sprintf("%s-%s-%s.terraform.tfvars.json", info.ContextPrefix, info.ComponentFolderPrefix, info.Component)
+	}
+
+	return varFile
+}
```

Comment:
The above two functions can be combined, no? I was thinking only the prefix would be returned depending on if the ComponentFolderPrefix

```
	var componentPrefix
	if len(info.ComponentFolderPrefix) == 0 {
		componentPrefix = fmt.Sprintf("%s-%s", info.ContextPrefix, info.Component)
	} else {
		componentPrefix = fmt.Sprintf("%s-%s-%s", info.ContextPrefix, info.ComponentFolderPrefix, info.Component)
	}
        return componentPrefix
```

then that function logic can be used for both the helmfile and terraform auto vars

Reply:
we can try to DRY everything if it makes sense.
But in many cases it's better to copy than to create abstraction on top of abstraction